source/mir/math/stat.d

```diff
@@ -420,6 +424,357 @@ unittest
     assert(float.max.repeat(3).hmean!(double, "fast").approxEqual(float.max));
 }
 
+///
+enum CentralTendency
+{
+    ///
+    mean,
+    hmean
+    //TODO:
+    ///
+    //median,
+    ///
+    //geometricMean,
+    ///
+    //weightedMean
+}
```

Comment:
Probably it would be better to implement `median` and `gmean` first, before we expose some more high-level API. This will help also to understand better do we need this high-level API and if we need it then to understand how it should look like.

`median` / `gmean`  would have other params if any.

`weightedMean` - I am not sure if it should be a part of this enum, instead of just using  `mean(r * weights)`.

Reply:
Plus things like normalization using standard deviation `(s - mean) * (1 / avg)`. First, we need to have all base building blocks.